MultiServer.py

```diff
@@ -955,18 +960,34 @@ def register_location_checks(ctx: Context, team: int, slot: int, locations: typi
     if new_locations:
         if count_activity:
             ctx.client_activity_timers[team, slot] = datetime.datetime.now(datetime.timezone.utc)
+
+        ctx.location_checks[team, slot] |= new_locations
+        slots_with_updated_hints: typing.Set[int] = set()
+        hint_locations_to_receiver: typing.Dict[int, int] = {
```

Comment:
This looks like a very serious performance degredation in one of our hottest code paths.
If you run `ctx.get_rechecked_hints(team, slot)` anyway, why even any of the surrounding code.

Reply:
I think ultimately, I may just change how hints are stored while in memory.